Modules/Core/Common/include/itkImage.hxx

```diff
@@ -148,7 +148,9 @@ Image<TPixel, VImageDimension>::GetNumberOfComponentsPerPixel() const
 {
   // use the GetLength() method which works with variable length arrays,
   // to make it work with as much pixel types as possible
-  const PixelType p{};
+  IndexType start;
+  start.Fill(0);
+  const PixelType p = this->GetPixel(start);
```

Comment:
Is the origin guaranteed to be a valid index into the image?  If not, `this->GetPixel(start)` may fail.

Reply:
If I understand correctly, indices are in image space, hence I'd dare to say that this would work. I have some doubts where it would fail (i.e. the call itself or when trying to get the pixel) should the image be a null pointer. Might be a good enough reason to think of alternatives.

Reply:
Generally, the `region` argument to `Image::SetRegions()` need not encompass the origin and if that is what is in effect before the call to `Image::Allocate()` then the origin would be a problem.  I don't know enough context for these edits to know whether this problem applies in this case.

Reply:
Thanks for the enlightening explanation @Leengit ! So enough reasons to go for another solution, probably what @N-Dekker proposed [here](https://github.com/InsightSoftwareConsortium/ITK/pull/2393#issuecomment-796582214).